// primereact/components/schedule/Schedule.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Schedule = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ObjectUtils = require('../utils/ObjectUtils');

var _ObjectUtils2 = _interopRequireDefault(_ObjectUtils);

var _jquery = require('jquery');

var _jquery2 = _interopRequireDefault(_jquery);

require('fullcalendar');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Schedule = exports.Schedule = function (_Component) {
    _inherits(Schedule, _Component);

    function Schedule() {
        _classCallCheck(this, Schedule);

        return _possibleConstructorReturn(this, (Schedule.__proto__ || Object.getPrototypeOf(Schedule)).apply(this, arguments));
    }

    _createClass(Schedule, [{
        key: 'gotoDate',
        value: function gotoDate(date) {
            this.schedule.fullCalendar('gotoDate', date);
        }
    }, {
        key: 'prev',
        value: function prev() {
            this.schedule.fullCalendar('prev');
        }
    }, {
        key: 'next',
        value: function next() {
            this.schedule.fullCalendar('next');
        }
    }, {
        key: 'prevYear',
        value: function prevYear() {
            this.schedule.fullCalendar('prevYear');
        }
    }, {
        key: 'nextYear',
        value: function nextYear() {
            this.schedule.fullCalendar('nextYear');
        }
    }, {
        key: 'today',
        value: function today() {
            this.schedule.fullCalendar('today');
        }
    }, {
        key: 'incrementDate',
        value: function incrementDate(duration) {
            this.schedule.fullCalendar('incrementDate', duration);
        }
    }, {
        key: 'changeView',
        value: function changeView(viewName) {
            this.schedule.fullCalendar('changeView', viewName);
        }
    }, {
        key: 'getDate',
        value: function getDate() {
            return this.schedule.fullCalendar('getDate');
        }
    }, {
        key: 'componentDidMount',
        value: function componentDidMount() {
            var _this2 = this;

            this.config = {
                theme: true,
                header: this.props.header,
                isRTL: this.props.rtl,
                weekends: this.props.weekends,
                hiddenDays: this.props.hiddenDays,
                fixedWeekCount: this.props.fixedWeekCount,
                weekNumbers: this.props.weekNumbers,
                businessHours: this.props.businessHours,
                height: this.props.height,
                contentHeight: this.props.contentHeight,
                aspectRatio: this.props.aspectRatio,
                eventLimit: this.props.eventLimit,
                defaultDate: this.props.defaultDate,
                timezone: this.props.timezone,
                editable: this.props.editable,
                droppable: this.props.droppable,
                eventStartEditable: this.props.eventStartEditable,
                eventDurationEditable: this.props.eventDurationEditable,
                defaultView: this.props.defaultView,
                allDaySlot: this.props.allDaySlot,
                allDayText: this.props.allDayText,
                slotDuration: this.props.slotDuration,
                slotLabelInterval: this.props.slotLabelInterval,
                snapDuration: this.props.snapDuration,
                scrollTime: this.props.scrollTime,
                minTime: this.props.minTime,
                maxTime: this.props.maxTime,
                slotEventOverlap: this.props.slotEventOverlap,
                nowIndicator: this.props.nowIndicator,
                dragRevertDuration: this.props.dragRevertDuration,
                dragOpacity: this.props.dragOpacity,
                dragScroll: this.props.dragScroll,
                eventOverlap: this.props.eventOverlap,
                eventConstraint: this.props.eventConstraint,
                eventRender: this.props.eventRender,
                dayRender: this.props.dayRender,
                dayClick: function dayClick(date, jsEvent, view) {
                    if (_this2.props.onDayClick) {
                        _this2.props.onDayClick({
                            'date': date,
                            'jsEvent': jsEvent,
                            'view': view
                        });
                    }
                },
                drop: function drop(date, jsEvent, ui, resourceId) {
                    if (_this2.props.onDrop) {
                        _this2.props.onDrop({
                            'date': date,
                            'jsEvent': jsEvent,
                            'resourceId': resourceId
                        });
                    }
                },
                eventClick: function eventClick(calEvent, jsEvent, view) {
                    if (_this2.props.onEventClick) {
                        _this2.props.onEventClick({
                            'calEvent': calEvent,
                            'jsEvent': jsEvent,
                            'view': view
                        });
                    }
                },
                eventMouseover: function eventMouseover(calEvent, jsEvent, view) {
                    if (_this2.props.onEventMouseover) {
                        _this2.props.onEventMouseover({
                            'calEvent': calEvent,
                            'jsEvent': jsEvent,
                            'view': view
                        });
                    }
                },
                eventMouseout: function eventMouseout(calEvent, jsEvent, view) {
                    if (_this2.props.onEventMouseout) {
                        _this2.props.onEventMouseout({
                            'calEvent': calEvent,
                            'jsEvent': jsEvent,
                            'view': view
                        });
                    }
                },
                eventDragStart: function eventDragStart(event, jsEvent, ui, view) {
                    if (_this2.props.onEventDragStart) {
                        _this2.props.onEventDragStart({
                            'event': event,
                            'jsEvent': jsEvent,
                            'view': view
                        });
                    }
                },
                eventDragStop: function eventDragStop(event, jsEvent, ui, view) {
                    if (_this2.props.onEventDragStop) {
                        _this2.props.onEventDragStop({
                            'event': event,
                            'jsEvent': jsEvent,
                            'view': view
                        });
                    }
                },
                eventDrop: function eventDrop(event, delta, revertFunc, jsEvent, ui, view) {
                    if (_this2.props.onEventDrop) {
                        _this2.props.onEventDrop({
                            'event': event,
                            'delta': delta,
                            'revertFunc': revertFunc,
                            'jsEvent': jsEvent,
                            'view': view
                        });
                    }
                },
                eventResizeStart: function eventResizeStart(event, jsEvent, ui, view) {
                    if (_this2.props.onEventResizeStart) {
                        _this2.props.onEventResizeStart({
                            'event': event,
                            'jsEvent': jsEvent,
                            'view': view
                        });
                    }
                },
                eventResizeStop: function eventResizeStop(event, jsEvent, ui, view) {
                    if (_this2.props.onEventResizeStop) {
                        _this2.props.onEventResizeStop({
                            'event': event,
                            'jsEvent': jsEvent,
                            'view': view
                        });
                    }
                },
                eventResize: function eventResize(event, delta, revertFunc, jsEvent, ui, view) {
                    if (_this2.props.onEventResize) {
                        _this2.props.onEventResize({
                            'event': event,
                            'delta': delta,
                            'revertFunc': revertFunc,
                            'jsEvent': jsEvent,
                            'view': view
                        });
                    }
                },
                viewRender: function viewRender(view, element) {
                    if (_this2.props.onViewRender) {
                        _this2.props.onViewRender({
                            'view': view,
                            'element': element
                        });
                    }
                }
            };

            if (this.props.locale) {
                for (var prop in this.props.locale) {
                    this.config[prop] = this.props.locale[prop];
                }
            }

            if (this.props.options) {
                for (var _prop in this.props.options) {
                    this.config[_prop] = this.props.options[_prop];
                }
            }

            this.schedule = (0, _jquery2.default)(this.scheduleEl);
            this.schedule.fullCalendar(this.config);
            this.events = [].concat(_toConsumableArray(this.props.events));
            this.schedule.fullCalendar('addEventSource', this.events);
        }
    }, {
        key: 'shouldComponentUpdate',
        value: function shouldComponentUpdate(nextProps, nextState) {
            return false;
        }
    }, {
        key: 'componentWillReceiveProps',
        value: function componentWillReceiveProps(nextProps) {
            if (!_ObjectUtils2.default.equals(nextProps.events, this.events)) {
                this.events = [].concat(_toConsumableArray(nextProps.events));
                this.schedule.fullCalendar('removeEventSources');
                this.schedule.fullCalendar('addEventSource', this.events);
            }
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            (0, _jquery2.default)(this.scheduleEl).fullCalendar('destroy');
        }
    }, {
        key: 'render',
        value: function render() {
            var _this3 = this;

            return _react2.default.createElement('div', { ref: function ref(el) {
                    return _this3.scheduleEl = el;
                }, style: this.props.style, className: this.props.styleClass });
        }
    }]);

    return Schedule;
}(_react.Component);

Schedule.defaultProps = {
    events: [],
    header: null,
    style: null,
    styleClass: null,
    isRTL: false,
    weekends: true,
    hiddenDays: [],
    fixedWeekCount: true,
    weekNumbers: false,
    businessHours: false,
    height: null,
    contentHeight: null,
    aspectRatio: 1.35,
    eventLimit: false,
    defaultDate: null,
    editable: false,
    droppable: false,
    eventStartEditable: true,
    eventDurationEditable: true,
    defaultView: 'month',
    allDaySlot: true,
    allDayText: 'all-day',
    slotDuration: '00:30:00',
    slotLabelInterval: null,
    snapDuration: null,
    scrollTime: '06:00:00',
    minTime: '00:00:00',
    maxTime: '24:00:00',
    slotEventOverlap: true,
    nowIndicator: false,
    dragRevertDuration: 500,
    dragOpacity: 0.75,
    dragScroll: true,
    eventOverlap: true,
    eventConstraint: null,
    eventRender: null,
    dayRender: null,
    locale: null,
    timezone: false,
    options: null,
    onDayClick: null,
    onEventClick: null,
    onEventMouseover: null,
    onEventMouseout: null,
    onEventDragStart: null,
    onEventDragStop: null,
    onEventDrop: null,
    onEventResizeStart: null,
    onEventResizeStop: null,
    onEventResize: null,
    onViewRender: null,
    onDrop: null
};
Schedule.propsTypes = {
    events: _react.PropTypes.array,
    header: _react.PropTypes.oneOfType([_react.PropTypes.object, _react.PropTypes.bool]),
    style: _react.PropTypes.string,
    styleClass: _react.PropTypes.string,
    isRTL: _react.PropTypes.bool,
    weekends: _react.PropTypes.bool,
    hiddenDays: _react.PropTypes.array,
    fixedWeekCount: _react.PropTypes.bool,
    weekNumbers: _react.PropTypes.bool,
    businessHours: _react.PropTypes.oneOfType([_react.PropTypes.bool, _react.PropTypes.object]),
    height: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.func, _react.PropTypes.string]),
    contentHeight: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.func, _react.PropTypes.string]),
    aspectRatio: _react.PropTypes.number,
    eventLimit: _react.PropTypes.oneOfType([_react.PropTypes.bool, _react.PropTypes.number]),
    defaultDate: _react.PropTypes.any,
    editable: _react.PropTypes.bool,
    droppable: _react.PropTypes.bool,
    eventStartEditable: _react.PropTypes.bool,
    eventDurationEditable: _react.PropTypes.bool,
    defaultView: _react.PropTypes.string,
    allDaySlot: _react.PropTypes.bool,
    allDayText: _react.PropTypes.string,
    slotDuration: _react.PropTypes.any,
    slotLabelInterval: _react.PropTypes.any,
    snapDuration: _react.PropTypes.any,
    scrollTime: _react.PropTypes.any,
    minTime: _react.PropTypes.any,
    maxTime: _react.PropTypes.any,
    slotEventOverlap: _react.PropTypes.bool,
    nowIndicator: _react.PropTypes.bool,
    dragRevertDuration: _react.PropTypes.number,
    dragOpacity: _react.PropTypes.number,
    dragScroll: _react.PropTypes.bool,
    eventOverlap: _react.PropTypes.oneOfType([_react.PropTypes.bool, _react.PropTypes.func]),
    eventConstraint: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.object]),
    eventRender: _react.PropTypes.func,
    dayRender: _react.PropTypes.func,
    locale: _react.PropTypes.object,
    timezone: _react.PropTypes.oneOfType([_react.PropTypes.bool, _react.PropTypes.string]),
    options: _react.PropTypes.object,
    onDayClick: _react.PropTypes.func,
    onEventClick: _react.PropTypes.func,
    onEventMouseover: _react.PropTypes.func,
    onEventMouseout: _react.PropTypes.func,
    onEventDragStart: _react.PropTypes.func,
    onEventDragStop: _react.PropTypes.func,
    onEventDrop: _react.PropTypes.func,
    onEventResizeStart: _react.PropTypes.func,
    onEventResizeStop: _react.PropTypes.func,
    onEventResize: _react.PropTypes.func,
    onViewRender: _react.PropTypes.func,
    onDrop: _react.PropTypes.func
};
